custom/enikshay/case_utils.py

```diff
@@ -170,6 +172,39 @@ def get_open_episode_case_from_person(domain, person_case_id):
     )
 
 
+def get_open_referral_case_from_person(domain, person_case_id):
+    episode = get_open_episode_case_from_person(domain, person_case_id)
+    case_accessor = CaseAccessors(domain)
+    reverse_indexed_cases = case_accessor.get_reverse_indexed_cases([episode.case_id])
+    open_referral_cases = [
+        case for case in reverse_indexed_cases
+        if not case.closed and case.type == CASE_TYPE_REFERRAL
+    ]
+    if not open_referral_cases:
+        return None
+    if len(open_referral_cases) == 1:
+        return open_referral_cases[0]
+    else:
+        raise ENikshayException(
+            "Expected none or one open referral case for person with id: {}".format(person_case_id)
+        )
+
+
+def get_latest_trail_case_from_person(domain, person_case_id):
+    episode = get_open_episode_case_from_person(domain, person_case_id)
+    case_accessor = CaseAccessors(domain)
+    reverse_indexed_cases = case_accessor.get_reverse_indexed_cases([episode.case_id])
+    trail_cases = [
+        case for case in reverse_indexed_cases
+        if case.type == CASE_TYPE_TRAIL
+    ]
+    trail_cases.sort(key=lambda c: c.opened_on)
```

Comment:
maybe have the sql function be parameterized by an order column since it's a pain to write those, but having a method that doesn't require you to pass in the order_by would be nice. minor nit: i think django normally uses `latest` rather than `newest`